utils_test.go

```diff
@@ -0,0 +1,16 @@
+package conntrack
+
+import (
+	"net/netip"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+)
+
+// Diff is a wrapper function around cmp.Diff. It helps make test cases less verbose by invoking
+// cmp.Diff with the desired options. In our case, we need cmpopts.EquateComparable(netip.Addr{}) to
+// make sure that structs which contain netip.Addr fields can be compared without a panic.
+// See https://github.com/google/go-cmp/issues/339
+func Diff(x, y interface{}) string {
```

Comment:
Honestly, not sure why I was using `cmp.Diff` to begin with. I think at the time assert.Equals' diff output was hard to understand, but nowadays it's just fine. Let's replace all uses of this with simple `assert.Equals` and remove our direct dependency on `go-cmp`.

Reply:
I agree. It wasn't "possible" when using `net.IP` as the semantics of `reflect.DeepEqual` for that type are different from the semantics of https://pkg.go.dev/net#IP.Equal (which is used by `cmp.Diff`).